### PR TITLE
Created SplitPane component

### PR DESCRIPTION
**Pull request by broncha**
Split Pane component was created for #1 

- ~~be able to split vertically or horizontally~~
- ~~have a percentage passed in that determines the relative sizes of the two panes~~
- ~~call a prop callback with any changes to the split percentage~~
- ~~have a nice draggable splitter line in between the parts~~
- ~~use flexbox for easy layout. We only support modern browsers in mWater~~

Currently the initial width is set in percentage. Once the resizing starts, the pane's width/height is set in pixels. Please check.

```
render: ->
    React.createElement(SplitPane, {split: "vertical", leftPaneSize: "20%", minLeftPaneSize: 100}, [
      H.div null
      React.createElement(SplitPane, {split: "horizontal", leftPaneSize: "20%", minLeftPaneSize: 100}, [
        H.div null
        React.createElement(SplitPane, {split: "vertical", leftPaneSize: "20%", minLeftPaneSize: 100}, [
          H.div null
          H.div null
        ])
      ])
    ])
```

### CODE REVIEW

File: src/Divider.coffee [hunk 1]
```diff
@@ -0,0 +1,23 @@
+React = require 'react'
+H = React.DOM
+
+module.exports = class Divider extends React.Component
+  constructor: ->
+    super
+    @state = { 
+      
+    }
```
Comment (grassick) [inline src/Divider.coffee]: The entire constructor can be eliminated when no state.

File: src/Pane.coffee [hunk 2]
```diff
@@ -0,0 +1,29 @@
+React = require 'react'
+H = React.DOM
+
+module.exports = class Pane extends React.Component
+  constructor: ->
+    super
+    @state = { 
+      
+    }
+
```
Comment (grassick) [inline src/Pane.coffee]: The entire constructor can be eliminated when no state.

File: src/Divider.coffee [hunk 3]
```diff
@@ -0,0 +1,23 @@
+React = require 'react'
+H = React.DOM
+
+module.exports = class Divider extends React.Component
```
Comment (grassick) [inline src/Divider.coffee]: Can you add a comment about what it does (short is ok).

File: src/Pane.coffee [hunk 4]
```diff
@@ -0,0 +1,29 @@
+React = require 'react'
+H = React.DOM
+
+module.exports = class Pane extends React.Component
+  constructor: ->
```
Comment (grassick) [inline src/Pane.coffee]: Needs `@propTypes` to describe what props it takes.

File: src/SplitPane.coffee [hunk 5]
```diff
@@ -0,0 +1,72 @@
+React = require 'react'
+H = React.DOM
+Pane = require './Pane'
+Divider = require './Divider'
+ReactDOM = require 'react-dom'
+
+module.exports = class SplitPane extends React.Component
```
Comment (grassick) [inline src/SplitPane.coffee]: Needs comments and `@propTypes` to describe what props it takes and thus how to use it.

**Comment by grassick** [comment]
Very nice professional job! Could you tidy up the few minor comments I made?

File: src/SplitPane.coffee [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    @state = { 
+      resizing: false
+      leftPaneSize: @props.leftPaneSize
+    }
+
+  @defaultProps: ->
+    split: 'vertical'
+
+  componentDidUpdate: =>
+    node = ReactDOM.findDOMNode(@refs.leftPane)
```
Comment (grassick) [inline src/SplitPane.coffee]: Entire function not needed, right?

File: src/SplitPane.coffee [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+Divider = require './Divider'
+ReactDOM = require 'react-dom'
+
+module.exports = class SplitPane extends React.Component
+  constructor: ->
+    super
+    @state = { 
+      resizing: false
+      leftPaneSize: @props.leftPaneSize
+    }
```
Comment (grassick) [inline src/SplitPane.coffee]: Maybe `firstPaneSize` since can be vertical?

**Comment by broncha** [comment]
@grassick Please check now.

**Comment by grassick** [comment]
Very nice work!

**Pull request merged**